src/app/views/sidebar/sample-queries/SampleQueries.tsx

```diff
@@ -70,19 +74,27 @@ export class SampleQueries extends Component<ISampleQueriesProps, any> {
   };
 
   public componentDidUpdate = (prevProps: ISampleQueriesProps) => {
+    const { samples: { queries }, permissionModeType } = this.props;
     const { sampleQueries } = this.state;
-    if (prevProps.samples.queries !== this.props.samples.queries) {
+    if (prevProps.permissionModeType !== permissionModeType || prevProps.samples.queries.sampleQueries !== queries.sampleQueries || prevProps.samples.queries.teamsAppSampleQueries !== queries.teamsAppSampleQueries) {
```

Comment:
what's happening here

Reply:
So the state of the component is only being refreshed if theres a state change in the redux props (sample queries pulled from devx api being different). The initial logic was quite simple because there was only one mode, so if the sample query lists werent the same that meant the sample queries needed to be refreshed. Now they are being refreshed if the application mode type has changed OR if theres any difference in the list

Reply:
This checking might have to be refined because the state (permissionModeType) is changing prematurely (when someone clicks a sample query, it toggles the permissionModeType), but I'm not sure where in the code clicking a sample query triggers the state change.

Reply:
That's a little bit concerning, I will dig deeper today